📌 Data-grid: make core-react peer dependency more robust

with `"@equinor/eds-core-react": "workspace:^"` in peerdependenices you will get npm ERR! ERESOLVE unable to resolve dependency tree. The peerdependency set to `^0.35.1` will error if actual version is` 0.36.0`. I think we need to use `>= `instead of `^`

--- a/packages/eds-data-grid-react/package.json
+++ b/packages/eds-data-grid-react/package.json
@@ -23,7 +23,7 @@
     "email": "fg_eds@equinor.com"
   },
   "peerDependencies": {
-    "@equinor/eds-core-react": "workspace:^",
+    "@equinor/eds-core-react": "workspace:>=0.36.0",
     "react": ">=16.8",
     "react-dom": ">=16.8",
     "styled-components": ">=4.2"

